pkg/sql/resolver.go

```diff
@@ -251,6 +251,36 @@ func (p *planner) IsTableVisible(
 	return false, true, nil
 }
 
+// IsTypeVisible is part of the tree.EvalDatabase interface.
+func (p *planner) IsTypeVisible(
+	ctx context.Context, curDB string, searchPath sessiondata.SearchPath, typeID oid.Oid,
+) (isVisible bool, exists bool, err error) {
+	// Check builtin types first. They are always globally visible.
+	if _, ok := types.OidToType[typeID]; ok {
+		return true, true, nil
+	}
+	// TODO(ajwerner): look at this error and only no-op if it is
```

Comment:
any reason we can't do this now

Reply:
doesn't seem there's a reason we can't do it now -- i meant to ask @ajwerner about this comment (which was just added recently). it does look like there's a few more errors than just `ErrDescriptorNotFound` that we need to check for

Reply:
i guess my concern is if it fails due to some transient error it will return False instead of error. arguably this is bad...

Reply:
Right, also, if it is a transaction restart I think it could expose some other anomalies. 

FWIW cleaning up the errors here and probably making these methods return clear booleans and deal with flags properly to declare desired behavior is a major roadmap item for schema this release. 

This code is actually a nightmare right now. I'm trying to untangle all of the errors you can get and it's making my angry. I'm 75% sure that the condition at the moment here is:

```
errors.Is(err, catalog.ErrDescriptorNotFound) || errors.Is(err, catalog.ErrDescriptorDropped) || pgcode.GetPGCode(err) == pgcode.UndefinedObject
```

Sorry :sweat:

Reply:
thank you for that list and the roadmap item!